trackers/react-native-tracker/src/subject.ts

```diff
@@ -0,0 +1,87 @@
+import { CorePluginConfiguration, PayloadBuilder, TrackerCore } from '@snowplow/tracker-core';
+import { ScreenSize, SubjectConfiguration } from './types';
+
+export function newSubject(core: TrackerCore, configuration?: SubjectConfiguration) {
+  let domainUserId: string | undefined;
+  let networkUserId: string | undefined;
+
+  const addSubjectToPayload = (payload: PayloadBuilder): void => {
+    payload.add('duid', domainUserId);
+    payload.add('nuid', networkUserId);
+  };
+
+  const setScreenResolution = (screenSize: ScreenSize) =>
+    core.setScreenResolution(String(screenSize[0]), String(screenSize[1]));
+
+  const setNetworkUserId = function (userId: string | undefined) {
+    networkUserId = userId;
+  };
+
+  const setDomainUserId = function (userId: string | undefined) {
+    domainUserId = userId;
+  };
+
+  const setColorDepth = function (colorDepth: number) {
+    core.setColorDepth(String(colorDepth));
+  };
+
+  const setScreenViewport = function (screenSize: ScreenSize) {
+    core.setViewport(String(screenSize[0]), String(screenSize[1]));
+  };
+
+  const setSubjectData = function (data: SubjectConfiguration) {
+    setNetworkUserId(data.networkUserId);
+    setDomainUserId(data.domainUserId);
+    if (data.userId) {
+      core.setUserId(data.userId);
+    }
+    if (data.useragent) {
+      core.setUseragent(data.useragent);
+    }
+    if (data.ipAddress) {
+      core.setIpAddress(data.ipAddress);
+    }
+    if (data.timezone) {
+      core.setTimezone(data.timezone);
+    }
+    if (data.language) {
+      core.setLang(data.language);
+    }
+    if (data.screenResolution) {
+      setScreenResolution(data.screenResolution);
+    }
+    if (data.colorDepth) {
+      setColorDepth(data.colorDepth);
+    }
+    if (data.screenViewport) {
+      setScreenViewport(data.screenViewport);
+    }
+  };
+
+  if (configuration) {
+    setSubjectData(configuration);
+  }
+
+  const subjectPlugin: CorePluginConfiguration = {
+    plugin: {
+      beforeTrack: addSubjectToPayload,
+    },
+  };
+
+  return {
+    subjectPlugin,
+    properties: {
+      setUserId: core.setUserId,
```

Comment:
There's probably a good reason, but why are we making these these `core.set*` functions available both here and as part of `setSubjectData`?

Reply:
It's mostly for user convenience so that they have an explicit setter for `setUserId` instead of the less intuitive `setSubjectData({userId: ..})`. Also trying to avoid a breaking change to come as close to the previous API as possible.